native/src/main/java/io/ballerina/stdlib/crypto/nativeimpl/Decode.java

```diff
@@ -133,28 +154,56 @@ public static Object decodeRsaPrivateKeyFromKeyFile(BString keyFilePath, Object
                         keyFilePath.getValue());
             }
             PrivateKey privateKey = converter.getPrivateKey(privateKeyInfo);
-            return buildPrivateKeyRecord(privateKey);
+            return buildRsPrivateKeyRecord(privateKey);
         } catch (FileNotFoundException e) {
             return CryptoUtils.createError("Key file not found at: " + privateKeyFile.getAbsoluteFile());
         } catch (PKCSException | IOException e) {
             return CryptoUtils.createError("Unable to do private key operations: " + e.getMessage());
         }
     }
 
-    private static Object buildPrivateKeyRecord(PrivateKey privateKey) {
+    private static Object buildRsPrivateKeyRecord(PrivateKey privateKey) {
         if (privateKey.getAlgorithm().equals(Constants.RSA_ALGORITHM)) {
-            BMap<BString, Object> privateKeyRecord = ValueCreator.
-                    createRecordValue(ModuleUtils.getModule(), Constants.PRIVATE_KEY_RECORD);
-            privateKeyRecord.addNativeData(Constants.NATIVE_DATA_PRIVATE_KEY, privateKey);
-            privateKeyRecord.put(StringUtils.fromString(Constants.PRIVATE_KEY_RECORD_ALGORITHM_FIELD),
-                                 StringUtils.fromString(privateKey.getAlgorithm()));
-            return privateKeyRecord;
+            return getPrivateKeyRecord(privateKey);
         } else {
-            return CryptoUtils.createError("Not a valid RSA key.");
+            return CryptoUtils.createError("Not a valid RSA key");
+        }
+    }
+
+    private static Object getPrivateKeyRecord(PrivateKey privateKey) {
+        BMap<BString, Object> privateKeyRecord = ValueCreator.
+                createRecordValue(ModuleUtils.getModule(), Constants.PRIVATE_KEY_RECORD);
+        privateKeyRecord.addNativeData(Constants.NATIVE_DATA_PRIVATE_KEY, privateKey);
+        privateKeyRecord.put(StringUtils.fromString(Constants.PRIVATE_KEY_RECORD_ALGORITHM_FIELD),
+                             StringUtils.fromString(privateKey.getAlgorithm()));
+        return privateKeyRecord;
+    }
+
+    private static Object buildEcPrivateKeyRecord(PrivateKey privateKey) {
+        if (privateKey.getAlgorithm().equals(Constants.EC_ALGORITHM)) {
+            return getPrivateKeyRecord(privateKey);
+        } else {
+            return CryptoUtils.createError("Not a valid EC key");
```

Comment:
can directly return here without else block

Reply:
Fixed in [6f1b573](https://github.com/ballerina-platform/module-ballerina-crypto/pull/533/commits/6f1b5734ec97d4e3a5a4e254f44fe11bf72a1202)